src/pybind/mgr/dashboard/tools.py

```diff
@@ -910,3 +915,31 @@ def find_object_in_list(key, value, iterable):
         if key in obj and obj[key] == value:
             return obj
     return None
+
+
+def update_dict(data, update_data):
```

Comment:
Calling the function with ``update_dict({'foo': {'bar': 5}}, {'xyz': 10})`` throws an error. I expect the result of ``{'foo': {'bar': 5}, 'xyz': 10}``.

Reply:
Can you ensure that the input parameters are always in the form which will not raise errors? If not, maybe it makes sense to implement a fully features dotted dictionary class like i did here: https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/lib/python3/dist-packages/openmediavault/collections.py#L57 and https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/share/openmediavault/unittests/python/test_openmediavault_collections.py

Reply:
Why?

Reply:
To make it right?